File: core/src/main/java/org/runetranscriber/core/viking/example/AngloSaxonExample5.java
package org.runetranscriber.core.viking.example;

import org.runetranscriber.core.AbstractExample;
import org.runetranscriber.core.FontLetterList;
import org.runetranscriber.core.LanguageLetterList;
import org.runetranscriber.core.Phoneme;
import org.runetranscriber.core.PhonemeList;
import org.runetranscriber.core.RuneList;
import org.runetranscriber.core.viking.AngloSaxonFontLetter;
import org.runetranscriber.core.viking.AngloSaxonRune;

/**
 * Provides an example for Anglo-Saxon runes from The Hobbit, Preface #1.
 */
public final class AngloSaxonExample5 extends AbstractExample<AngloSaxonRune, AngloSaxonFontLetter>
{
    @Override
    protected FontLetterList<AngloSaxonFontLetter> createFontLetters()
    {
        final AngloSaxonFontLetter[] myFontLetters = {
                // the
                AngloSaxonFontLetter.THORN,
                AngloSaxonFontLetter.EH,
                AngloSaxonFontLetter.SPACE,
                // hobbit
                AngloSaxonFontLetter.HAEGL,
                AngloSaxonFontLetter.OS,
                AngloSaxonFontLetter.BEORC,
                AngloSaxonFontLetter.BEORC,
                AngloSaxonFontLetter.IS,
                AngloSaxonFontLetter.TIW,
                AngloSaxonFontLetter.NEWLINE,
                // or
                AngloSaxonFontLetter.OS,
                AngloSaxonFontLetter.RAD,
                AngloSaxonFontLetter.NEWLINE,
                // there
                AngloSaxonFontLetter.THORN, AngloSaxonFontLetter.EH,
                AngloSaxonFontLetter.RAD,
                AngloSaxonFontLetter.EH,
                AngloSaxonFontLetter.SPACE,
                // and
                AngloSaxonFontLetter.AESC, AngloSaxonFontLetter.NYD,
                AngloSaxonFontLetter.DAEG,
                AngloSaxonFontLetter.SPACE,
                // back
                AngloSaxonFontLetter.BEORC, AngloSaxonFontLetter.AESC, AngloSaxonFontLetter.CEN,
                AngloSaxonFontLetter.CEN, AngloSaxonFontLetter.SPACE,
                // again
                AngloSaxonFontLetter.AESC, AngloSaxonFontLetter.GYFU, AngloSaxonFontLetter.AESC,
                AngloSaxonFontLetter.IS, AngloSaxonFontLetter.NYD, };

        return new FontLetterList<AngloSaxonFontLetter>(myFontLetters);
    }

    @Override
    protected LanguageLetterList createLanguageLetters()
    {
        final String text = "The hobbit\nor\nthere and back again";

        return new LanguageLetterList(text);
    }

    /**
     * @return phonemes.
     */
    @Override
    protected PhonemeList createPhonemes()
    {
        final Phoneme[] myPhonemes = {
                // The
                Phoneme.TH, Phoneme.E, Phoneme.SPACE,
                // hobbit
                Phoneme.H, Phoneme.O, Phoneme.B, Phoneme.B, Phoneme.I, Phoneme.T, Phoneme.NEWLINE,
                // or
                Phoneme.O, Phoneme.R, Phoneme.NEWLINE,
                // there
                Phoneme.TH, Phoneme.E, Phoneme.R, Phoneme.E, Phoneme.SPACE,
                // and
                Phoneme.AE, Phoneme.N, Phoneme.D, Phoneme.SPACE,
                // back
                Phoneme.B, Phoneme.AE, Phoneme.K, Phoneme.K, Phoneme.SPACE,
                // again
                Phoneme.AE, Phoneme.G, Phoneme.AE, Phoneme.I, Phoneme.N, };

        return new PhonemeList(myPhonemes);
    }

    /**
     * @return runes.
     */
    @Override
    protected RuneList<AngloSaxonRune> createRunes()
    {
        final AngloSaxonRune[] myRunes = {
                // the
                AngloSaxonRune.THORN, AngloSaxonRune.EH,
                AngloSaxonRune.SPACE,
                // hobbit
                AngloSaxonRune.HAEGL, AngloSaxonRune.OS, AngloSaxonRune.BEORC, AngloSaxonRune.BEORC, AngloSaxonRune.IS,
                AngloSaxonRune.TIW, AngloSaxonRune.NEWLINE,
                // or
                AngloSaxonRune.OS, AngloSaxonRune.RAD, AngloSaxonRune.NEWLINE,
                // there
                AngloSaxonRune.THORN, AngloSaxonRune.EH, AngloSaxonRune.RAD, AngloSaxonRune.EH, AngloSaxonRune.SPACE,
                // and
                AngloSaxonRune.AESC, AngloSaxonRune.NYD, AngloSaxonRune.DAEG, AngloSaxonRune.SPACE,
                // back
                AngloSaxonRune.BEORC, AngloSaxonRune.AESC, AngloSaxonRune.CEN, AngloSaxonRune.CEN,
                AngloSaxonRune.SPACE,
                // again
                AngloSaxonRune.AESC, AngloSaxonRune.GYFU, AngloSaxonRune.AESC, AngloSaxonRune.IS, AngloSaxonRune.NYD, };

        return new RuneList<AngloSaxonRune>(myRunes);
    }
}

File: webapp/src/main/js/example/AngloSaxonExample5.js
/*
 * Provides an example for Anglo-Saxon runes from The Hobbit, Preface #1.
 */
define(
        [ "AngloSaxonRune" ],
        function(AngloSaxonRune)
        {
            "use strict";
            var AngloSaxonExample5 =
            {
                NAME: "Anglo-Saxon Example 5",

                IMAGE: "https://rawgit.com/jmthompson2015/runetranscriber/master/core/doc/images/hobbit/TheHobbit_Preface1.jpg",

                LANGUAGE_STRING: "The hobbit\nor\nthere and back again",

                LANGUAGE_WORDS: [ "the", " ", "hobbit", "\n", "or", "\n", "there", " ", "and", " ", "back", " ",
                        "again" ],

                PHONEMES: [ AngloSaxonRune.properties["thorn"].phoneme, AngloSaxonRune.properties["eh"].phoneme,
                        AngloSaxonRune.properties["space"].phoneme, AngloSaxonRune.properties["haegl"].phoneme,
                        AngloSaxonRune.properties["os"].phoneme, AngloSaxonRune.properties["beorc"].phoneme,
                        AngloSaxonRune.properties["beorc"].phoneme, AngloSaxonRune.properties["is"].phoneme,
                        AngloSaxonRune.properties["tiw"].phoneme, AngloSaxonRune.properties["newline"].phoneme,
                        AngloSaxonRune.properties["os"].phoneme, AngloSaxonRune.properties["rad"].phoneme,
                        AngloSaxonRune.properties["newline"].phoneme, AngloSaxonRune.properties["thorn"].phoneme,
                        AngloSaxonRune.properties["eh"].phoneme, AngloSaxonRune.properties["rad"].phoneme,
                        AngloSaxonRune.properties["eh"].phoneme, AngloSaxonRune.properties["space"].phoneme,
                        AngloSaxonRune.properties["aesc"].phoneme, AngloSaxonRune.properties["nyd"].phoneme,
                        AngloSaxonRune.properties["daeg"].phoneme, AngloSaxonRune.properties["space"].phoneme,
                        AngloSaxonRune.properties["beorc"].phoneme, AngloSaxonRune.properties["aesc"].phoneme,
                        AngloSaxonRune.properties["cen"].phoneme, AngloSaxonRune.properties["cen"].phoneme,
                        AngloSaxonRune.properties["space"].phoneme, AngloSaxonRune.properties["aesc"].phoneme,
                        AngloSaxonRune.properties["gyfu"].phoneme, AngloSaxonRune.properties["aesc"].phoneme,
                        AngloSaxonRune.properties["is"].phoneme, AngloSaxonRune.properties["nyd"].phoneme ],

                RUNES: [
                // the
                AngloSaxonRune.THORN, AngloSaxonRune.EH, AngloSaxonRune.SPACE,
                // hobbit
                AngloSaxonRune.HAEGL, AngloSaxonRune.OS, AngloSaxonRune.BEORC, AngloSaxonRune.BEORC, AngloSaxonRune.IS,
                        AngloSaxonRune.TIW, AngloSaxonRune.NEWLINE,
                        // or
                        AngloSaxonRune.OS, AngloSaxonRune.RAD, AngloSaxonRune.NEWLINE,
                        // there
                        AngloSaxonRune.THORN, AngloSaxonRune.EH, AngloSaxonRune.RAD, AngloSaxonRune.EH,
                        AngloSaxonRune.SPACE,
                        // and
                        AngloSaxonRune.AESC, AngloSaxonRune.NYD, AngloSaxonRune.DAEG, AngloSaxonRune.SPACE,
                        // back
                        AngloSaxonRune.BEORC, AngloSaxonRune.AESC, AngloSaxonRune.CEN, AngloSaxonRune.CEN,
                        AngloSaxonRune.SPACE,
                        // again
                        AngloSaxonRune.AESC, AngloSaxonRune.GYFU, AngloSaxonRune.AESC, AngloSaxonRune.IS,
                        AngloSaxonRune.NYD, ],

                FONT_LETTERS: [ AngloSaxonRune.properties["thorn"].fontLetter,
                        AngloSaxonRune.properties["eh"].fontLetter, AngloSaxonRune.properties["space"].fontLetter,
                        AngloSaxonRune.properties["haegl"].fontLetter, AngloSaxonRune.properties["os"].fontLetter,
                        AngloSaxonRune.properties["beorc"].fontLetter, AngloSaxonRune.properties["beorc"].fontLetter,
                        AngloSaxonRune.properties["is"].fontLetter, AngloSaxonRune.properties["tiw"].fontLetter,
                        AngloSaxonRune.properties["newline"].fontLetter, AngloSaxonRune.properties["os"].fontLetter,
                        AngloSaxonRune.properties["rad"].fontLetter, AngloSaxonRune.properties["newline"].fontLetter,
                        AngloSaxonRune.properties["thorn"].fontLetter, AngloSaxonRune.properties["eh"].fontLetter,
                        AngloSaxonRune.properties["rad"].fontLetter, AngloSaxonRune.properties["eh"].fontLetter,
                        AngloSaxonRune.properties["space"].fontLetter, AngloSaxonRune.properties["aesc"].fontLetter,
                        AngloSaxonRune.properties["nyd"].fontLetter, AngloSaxonRune.properties["daeg"].fontLetter,
                        AngloSaxonRune.properties["space"].fontLetter, AngloSaxonRune.properties["beorc"].fontLetter,
                        AngloSaxonRune.properties["aesc"].fontLetter, AngloSaxonRune.properties["cen"].fontLetter,
                        AngloSaxonRune.properties["cen"].fontLetter, AngloSaxonRune.properties["space"].fontLetter,
                        AngloSaxonRune.properties["aesc"].fontLetter, AngloSaxonRune.properties["gyfu"].fontLetter,
                        AngloSaxonRune.properties["aesc"].fontLetter, AngloSaxonRune.properties["is"].fontLetter,
                        AngloSaxonRune.properties["nyd"].fontLetter ],
            };

            if (Object.freeze)
            {
                Object.freeze(AngloSaxonExample5);
            }

            return AngloSaxonExample5;
        });
